lib/gitsh/git_repository.rb

```diff
@@ -62,8 +62,10 @@ def config(name, default = nil, force_default_git_command = false)
       out, err, status = Open3.capture3(command)
       if status.success?
         out.chomp
+      elsif block_given?
+        yield
       else
-        default
+        raise KeyError, "Git configuration variable #{name} is not set"
```

Comment:
Yes, it's mostly to make the message clear: this is where the exception comes from originally with a message indicating that the variable we're looking for didn't match any Git configuration variables, but it's re-raised in `Environment#fetch` with a different message to make it clear that the variable didn't exist anywhere we looked for it.